From f700975f514758a4a7582c2c972fa54ecfca3302 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 25 Jan 2018 17:55:02 +0200
Subject: [PATCH 1/4] gst/gstpluginloader.c: when env var is set do not fall
 through to system plugin scanner

If we set a custom GST_PLUGIN_SCANNER env var, then we probably want to use that and only that.

Falling through to the one installed on the system is problamatic in cross-compilation
environemnts, regardless of whether one pointed to by the env var succeeded or failed.

written by: Alexander Kanavin <alex.kanavin@gmail.com>

Upstream-Status: Submitted [https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/669]
Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>
---
 gst/gstpluginloader.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gst/gstpluginloader.c b/gst/gstpluginloader.c
index d1e404d..2e00c3c 100644
--- a/gst/gstpluginloader.c
+++ b/gst/gstpluginloader.c
@@ -475,9 +475,7 @@ gst_plugin_loader_spawn (GstPluginLoader * loader)
     helper_bin = g_strdup (env);
     res = gst_plugin_loader_try_helper (loader, helper_bin);
     g_free (helper_bin);
-  }
-
-  if (!res) {
+  } else {
     GST_LOG ("Trying installed plugin scanner");
 
 #ifdef G_OS_WIN32
-- 
2.28.0

